wordpress-mobile/WordPress-Android
# Manual feature configuration

This PR adds functionality to manually override any local or remote feature flag. I know this PR is big but as this feature will only ever be used by us or testers, I thought it would be OK (but I can still split the PR into more if necessary). 

I've added a `ManualFeatureConfig` which saves the locally used feature state into apps preferences. This functionality is behind a flag which is available in wasabi and jalapeno. The main reason is that we want to be able to test a developed feature on a PR build. I tried to minimize the amount of changes by reusing existing styles/dimens (cause the UI of this screen doesn't really matter).

The new screen can be found in Me/App settings/Test feature configuration if you're on Jalapeno or Wasabi builds. The screen has 2 sections. One for the remote features (the list is generated from the feature config files annotated with the `Feature` annotation) and the other for features in development (the list is generated from the feature config files annotated with the new `FeatureInDevelopment` annotation). You might notice that the features in development start in a `?` state. The reason is that we only show the manually overridden value there and there is no way to find out the current build config value without affecting the dev comfort while creating the flags.

I've added a "Restart" button which appears after any change. The reason is that the changes to flags sometimes need the app to be restarted to appear. The button should nudge the devs/testers to do so. 

![Screenshot_1600173924](https://user-images.githubusercontent.com/1079756/93212818-649c2880-f763-11ea-93e7-c3110dd14902.png)


To test:
- Go to Me/App settings/Test feature configuration
- Notice the new screen
- Notice that the remote configs are set based on the remote value
- Notice that the features in development are all `?`
- Try enabling/disabling features
- Check that this state is stored and survives app restart
- Check that the state has impact on the features
- Check that the restart button appears on any change

PR submission checklist:
